clippy_lints/src/question_mark.rs

```diff
@@ -231,6 +257,11 @@ fn expr_return_none_or_err(
 }
 
 impl<'tcx> LateLintPass<'tcx> for QuestionMark {
+    fn check_stmt(&mut self, cx: &LateContext<'tcx>, stmt: &'tcx Stmt<'_>) {
+        if !in_constant(cx, stmt.hir_id) {
+            check_let_some_else_return_none(cx, stmt);
```

Comment:
I think this could also check whether `question_mark` is allowed or not for the given stmt, if it is, it could lint